tests/phpunit/includes/build-tree-representation.php

```diff
@@ -0,0 +1,264 @@
+<?php declare( strict_types = 1 );
+
+/* phpcs:disable WordPress.Security.EscapeOutput.ExceptionNotEscaped */
+
+/**
+ * Generates the tree-like structure represented in the Html5lib tests.
+ *
+ * @see https://github.com/html5lib/html5lib-tests/blob/a9f44960a9fedf265093d22b2aa3c7ca123727b9/tree-construction/README.md
+ *
+ * @param string|null $fragment_context Context element in which to parse HTML, such as BODY or SVG.
+ * @param string      $html             Given test HTML.
+ * @return string|null Tree structure of parsed HTML, if supported, else null.
+ */
+function build_tree_representation( string $html, ?string $fragment_context ) {
```

Comment:
This function could be potentially moved into `tests/phpunit/includes/utils.php`.

Reply:
also its name is widely general and could benefit from narrowing. perhaps something like `html_debug_as_tree_string()` or even just `html_debug_as_tree()` or `build_equivalent_html_semantic_tree()`?

Reply:
True! I like the last suggestion the most: e8ceff2d40